source/MRMesh/MRVector3.h

```diff
@@ -43,7 +44,13 @@ struct Vector3
     constexpr       T & operator []( int e )       noexcept { return *( &x + e ); }
 
     T lengthSq() const { return x * x + y * y + z * z; }
-    T length() const { return std::sqrt( lengthSq() ); }
+    auto length() const
+    {
+        // Calling `sqrt` this way to hopefully support boost.multiprecision numbers.
+        // Returning `auto` to not break on integral types.
+        using std::sqrt;
+        return sqrt( lengthSq() );
```

Comment:
why should we support it here if we dont call this function for multipresion numbers?

Reply:
Since it's easy to support multiprecision numbers here, I think it's better to support them rather than artificially disable the support via `requires`.